modules/cms/classes/Router.php

```diff
@@ -300,15 +300,13 @@ public function getUrl()
 
     /**
      * Returns a routing parameter.
-     * @return array
+     * @param  string $name
+     * @param  string|null $default
+     * @return string|null
      */
-    public function getParameter($name, $default = null)
+    public function getParameter(string $name, string $default = null)
     {
-        if (isset($this->parameters[$name]) && ($this->parameters[$name] === '0' || !empty($this->parameters[$name]))) {
-            return $this->parameters[$name];
-        }
-
-        return $default;
+        return $this->parameters[$name] ?? $default;
```

Comment:
@martendej I think it is possible for parameters to be provided as an empty string depending on routing rules in place, eg. optional parameters - previously, this would use the default value. This change will prevent that from happening. Would you be able to revert it to using the condition block that was there before? The typehint changes above are fine to leave in.

Reply:
I changed the code to account for this situation, can you check if this is ok with you?

Reply:
Thanks @martendej.